cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -106,25 +106,37 @@ Result<compute::Declaration> FromProto(const substrait::Rel& rel,
           path = item.uri_path_glob();
         }
 
+        ::arrow::internal::Uri uri;
+        RETURN_NOT_OK(uri.Parse(path));
         if (item.format() ==
             substrait::ReadRel::LocalFiles::FileOrFiles::FILE_FORMAT_PARQUET) {
           format = std::make_shared<dataset::ParquetFileFormat>();
-        } else if (util::string_view{path}.ends_with(".arrow")) {
+        } else if (uri.extension() == ".arrow") {
```

Comment:
we may also want to support ".arrows"?

Reply:
I suggested it mostly for completeness, I don't think it's particularly likely though.

Reply:
If the situation arises, it can be added to the Substrait spec in a backwards-compatible way.